src/mesh/p4est_mesh.jl

```diff
@@ -199,14 +194,21 @@ function P4estMesh(meshfile::String;
                                                 n_trees)
   calc_tree_node_coordinates!(tree_node_coordinates, nodes, mapping, vertices, tree_to_vertex)
 
-  p4est = p4est_new_ext(0, conn, 0, initial_refinement_level, true, 0, C_NULL, C_NULL)
-  ghost = p4est_ghost_new(p4est, P4EST_CONNECT_FACE)
-  p4est_mesh = p4est_mesh_new(p4est, ghost, P4EST_CONNECT_FACE)
+  # Use Int-Vector of size 2 as quadrant user data
+  p4est = p4est_new_ext(0, conn, 0, initial_refinement_level, true, 2 * sizeof(Int), C_NULL, C_NULL)
+
+  # Destroy p4est structs at exit of Julia
+  function destroy_p4est_structs()
+    p4est_destroy(p4est)
+    p4est_connectivity_destroy(conn)
+  end
+
+  atexit(destroy_p4est_structs)
```

Comment:
In the docs, it says "The type of `x` must be a `mutable struct`, otherwise the behavior of this function is unpredictable." Can I still use this here?

Reply:
Don't these p4est types belong to the `mutable struct P4estMesh`?

Reply:
Ah yes, sorry. I was a bit confused.